cfme/fixtures/networks.py

```diff
@@ -29,18 +29,20 @@ def test_something(targeted_refresh)
                     targeted_refresh.register_target('ref-aaaabb', 'Router named TEST')
     """
     evm_log = '/var/www/miq/vmdb/log/evm.log'
-    needle_template = r'^.*Collection of targets with id.*:ems_ref=>"{}".*$'
-    merkyl_inspector.add_log(evm_log)
-    merkyl_inspector.reset_log(evm_log)
-    targets = set()  # { (ems_ref, comment), (ems_ref, comment), ... }
+    needle_template = r'.*Collection of targets with id.*:ems_ref=>"{}".*'
+    targets = {}
+    log_validator = LogValidator(evm_log)
+    log_validator.fix_before_start()
 
     def check_log():
-        logger.info('Looking for %s needles in evm.log: %s', len(targets), [t[1] for t in targets])
-        content = merkyl_inspector.get_log(evm_log)
-        for target in set(targets):
-            if target[0].search(content):
-                logger.info('Found needle %s', target[1])
-                targets.remove(target)
+        logger.info('Looking for %s needles in evm.log: %s' % (
+            len(targets), list(targets.values())))
+        log_validator.update_matched_patterns(list(targets.keys()))
+        found_targets = log_validator.validate_logs(_auto_fail=False)
+        if len(found_targets) > 0:
```

Comment:
No need for this check. If an empty list will be returned, then block won't be executed anyway.